### PR TITLE
BREAKING CHANGE(*): Remove dependency on Ergo. Remove Draft, Parse, Initialize, Trigger from API

### PR DESCRIPTION
**Pull request by mttrbrts**
# Closes #<CORRESPONDING ISSUE NUMBER>

### Changes

- Removes all dependencies on `@accordproject/ergo-compiler` and `@accordproject/ergo-engine`
- Removes `cicero-engine`, `cicero-server` and `cicero-test` packages which all deal with logic operations. This will be replaced by an Accord Project Agreement Protocol interface in future. Removed corresponding `initialize` and `trigger` commands.
- Removes `draft`, `normalize` and `parse` commands, these will be delegated to the new Template Engine in future.

### Flags

- The behavior of getRequestModels, getResponseModels, getStateModels & getEmitModels has changed to list the models in the template that are subtypes of Request, Response, State and Obligation respectively. The previous behaviour required introspection of the source code to discover the function signature. 

### Related Issues
- Issue #<NUMBER>
- Pull Request #<NUMBER>

### Author Checklist
- [ ] Ensure you provide a [DCO sign-off](https://github.com/probot/dco#how-it-works) for your commits using the `--signoff` option of git commit.
- [ ] Vital features and changes captured in unit and/or integration tests
- [ ] Commits messages follow [AP format](https://github.com/accordproject/techdocs/blob/master/DEVELOPERS.md#commit-message-format)
- [ ] Extend the documentation, if necessary
- [ ] Merging to `master` from `fork:branchname`

### CODE REVIEW

File: packages/cicero-core/src/templateloader.js [hunk 1]
```diff
@@ -238,22 +203,193 @@ class TemplateLoader extends FileLoader {
...
+        return result.reduce((a, f) => a.concat(f), []).filter((f) => f !== null);
     }
 
     /**
      * Prepare the text for parsing (normalizes new lines, etc)
      * @param {string} input - the text for the clause
      * @return {string} - the normalized text for the clause
      */
-    static normalizeText(input) {
+    static normalizeNLs(input) {
```
Comment (dselman) [inline packages/cicero-core/src/templateloader.js:392]: Can remove?

File: packages/cicero-core/test/data/verifying-template-signature/helloworldstateSigned/package.json [hunk 2]
```diff
@@ -1 +1 @@
...
\ No newline at end of file
+{"name":"helloworldstate","displayName":"Hello World State","version":"0.15.0","description":"This is the stateful Hello World of Accord Project Templates. Executing the clause will simply echo back the text that occurs after the string `Hello` prepended to text that is passed in the request along with the number of times the clause has been called.","author":"Accord Project","license":"Apache-2.0","accordproject":{"template":"clause","cicero":"^0.24.0","runtime":"cicero"},"devDependencies":{"cucumber":"^5.1.0"},"scripts":{"test":"cucumber-js test -r .cucumber.js"},"keywords":["hello","world","greet"]}
```
Comment (dselman) [inline packages/cicero-core/test/data/verifying-template-signature/helloworldstateSigned/package.json:1]: I assume the cucumber stuff is now dead?

**Review by dselman — CHANGES_REQUESTED** [review]
Nice work! A few questions.

File: packages/cicero-core/src/apmodelmanager.js [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+     */
+    constructor() {
+        super();
+        this.addCTOModel(Builtin.TimeModel, '@models.accordproject.org.time@0.2.0.cto');
+        this.addCTOModel(Builtin.MoneyModel, '@models.accordproject.org.accordproject.money@0.2.0.cto');
+        this.addCTOModel(Builtin.ContractModel, '@models.accordproject.org.accordproject.contract.cto');
+        this.addCTOModel(Builtin.RuntimeModel, '@models.accordproject.org.accordproject.runtime.cto');
+        this.addCTOModel(Builtin.OptionsModel, '@org.accordproject.ergo.options.cto');
+        this.validateModelFiles();
+        this.builtInNamespaces = this.getNamespaces();
```
[the remark this answers is not in the document]
Comment (mttrbrts) [reply packages/cicero-core/src/apmodelmanager.js:62]: This file is blindly copied from ergo-runtime.

All fair comments.

File: packages/cicero-core/src/builtin.js
[the remark this answers is not in the document]
Comment (mttrbrts) [reply packages/cicero-core/src/builtin.js:1]: Again, blindly copied from ergo-runtime to make the tests pass before further refactoring. We can likely do without

File: packages/cicero-core/test/data/verifying-template-signature/helloworldstateSigned/package.json (hunk 2 above)
Comment (mttrbrts) [reply packages/cicero-core/test/data/verifying-template-signature/helloworldstateSigned/package.json:1]: Yep, cicero-test is gone

File: packages/cicero-core/src/template.js [hunk 4]
```diff
@@ -50,15 +49,15 @@ class Template {
      */
     constructor(packageJson, readme, samples, request, logo, options, authorSignature) {
         this.metadata = new Metadata(packageJson, readme, samples, request, logo);
-        this.logicManager = new LogicManager('es6', null, options);
+        this.modelManager = new APModelManager();
+        this.introspector = new Introspector(this.modelManager);
         const templateKind = this.getMetadata().getTemplateType() !== 0 ? 'clause' : 'contract';
         this.parserManager = new ParserManager(this.getModelManager(),null,templateKind);
```
[the remark this answers is not in the document]
Comment (mttrbrts) [reply packages/cicero-core/src/template.js:55]: We rely on the ParserManager to identify the template grammar. This is likely overkill now, however, I don't want to introduce a new dependency on the Template Engine in this PR.

File: packages/cicero-core/test/data/verifying-template-signature/helloworldstateSigned/package.json (hunk 2 above)
Comment (mttrbrts) [reply packages/cicero-core/test/data/verifying-template-signature/helloworldstateSigned/package.json:1]: I'll remove in a future PR.

**Review by dselman — APPROVED** [review]
Didn't have time to review, sorry - but don't want to block!

**Pull request merged**